modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -192,26 +201,69 @@ public void setCopyrights(String copyrights) {
+                case SW360Attributes.RELEASE_EXTERNAL_ID_OREPO:
+                    releaseTagUrl = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_SWHID:
+                    softwareHeritageId = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_CHANGESTATUS:
+                    changeStatus = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_COPYRIGHTS:
+                    copyrights = value;
+                    break;
+                default:
+                    if (key.startsWith(SW360Attributes.RELEASE_EXTERNAL_ID_HASHES)) {
+                        if(hashes == null) {
+                            hashes = Collections.singleton(value);
+                        } else {
+                            hashes.add(value);
+                        }
+                    } else if (SW360CoordinateKeysToArtifactCoordinates.getValues().contains(key)) {
+                        if(coordinates == null) {
+                            coordinates = new HashMap<String, String>() {{
+                                put(key, value);
+                            }};
+                        } else {
+                            coordinates.put(key, value);
+                        }
+                    }
+            }
+        });
+        return this;
     }
 
     public SW360Release mergeWith(SW360Release releaseWithPrecedence) {
-        cpeid = Optional.of(releaseWithPrecedence.getCpeid()).orElse(cpeid);
-        downloadurl = Optional.of(releaseWithPrecedence.getDownloadurl()).orElse(downloadurl);
-        mainLicenseIds = Optional.of(releaseWithPrecedence.getMainLicenseIds()).orElse(mainLicenseIds);
-        coordinates = Optional.of(releaseWithPrecedence.getCoordinates()).orElse(coordinates);
-        finalLicense = Optional.of(releaseWithPrecedence.getFinalLicense()).orElse(finalLicense);
-        declaredLicense = Optional.of(releaseWithPrecedence.getDeclaredLicense()).orElse(declaredLicense);
-        observedLicense = Optional.of(releaseWithPrecedence.getObservedLicense()).orElse(observedLicense);
-        releaseTagUrl = Optional.of(releaseWithPrecedence.getReleaseTagUrl()).orElse(releaseTagUrl);
-        softwareHeritageId = Optional.of(releaseWithPrecedence.getSoftwareHeritageId()).orElse(softwareHeritageId);
-        hashes = Optional.of(releaseWithPrecedence.getHashes()).orElse(hashes);
-        clearingState = Optional.of(releaseWithPrecedence.getClearingState()).orElse(clearingState);
-        changeStatus = Optional.of(releaseWithPrecedence.getChangeStatus()).orElse(changeStatus);
-        copyrights = Optional.of(releaseWithPrecedence.getCopyrights()).orElse(copyrights);
-        externalIds = Optional.of(releaseWithPrecedence.getExternalIds()).orElse(externalIds);
+        cpeid = Optional.ofNullable(releaseWithPrecedence.getCpeid()).orElse(Optional.ofNullable(cpeid).orElse(null));
```

Comment:
every line has the same pattern, maybe extract into a function:
```
    private T fun(SW360Release release, Function<SW360Release,T> getter);
```

Then you would only have to write `cpeid = fun(releaseWithPrecedence, SW360Release::getCpeId)` which is cleaner and more readable

Reply:
The call to `Optional.ofNullable(cpeid).orElse(null)` should be a no-op and always equal to `cpeid`, i.e.
- if `cpeid` is not null, it just returns `cpeid`
- if `cpeid` is null, it falls in the `orElse` case and returns `null` which is still equal to `cpeid`

Reply:
when I put `orElse(cpeid)` in there, it caused a null pointer excpetion when it wasn't set. Hence I changed it to this mess, that worked.